bin/setup-radiopadre-virtualenv

```diff
@@ -0,0 +1,238 @@
+    else:
+        message(f"Trying to download CARTA via {CARTA_LINK}")
+        subprocess.check_call(f"cd {PADRE_PATH}; wget {CARTA_LINK}", shell=True)
+        if not os.path.exists(carta_tarball_path):
+            bye(f"{carta_tarball} failed to download")
+    message(f"Unpacking CARTA tarball")
+    subprocess.check_call(f"cd {PADRE_PATH}; tar zxvf {carta_tarball}", shell=True)
+    message(f"Linking {carta} to {carta_dir}")
+    os.symlink(carta_dir, carta)
+
+if os.path.exists(carta):
+    carta_link = PADRE_VENV + "/carta"
+    message(f"Linking {carta_link} to {carta}")
+    if os.path.exists(carta_link):
+        os.unlink(carta_link)
+    os.symlink(carta, carta_link)
+else:
+    message("No CARTA installation found")
+
+# install JS9
+if not options.no_js9:
+    js9status = open(PADRE_VENV + "/js9status", "w")
+
+    # install JS9 inside venv
+    # determine whether to use existing directory or repo
+    js9_git = js9_dir = js9_temp = None
+    if options.js9:
+        if options.js9 == 'git':
+            js9_git = JS9_REPO
+            message("Will clone JS9 from default location ({JS9_REPO})")
+        elif options.js9.split(":")[0] in ["http", "git"]:
+            js9_git = options.js9
+            message("Will clone JS9 from {js9_git}")
+        elif os.path.isdir(options.js9):
+            js9_dir = options.js9
+            message("Will install JS9 from local directory {js9_dir}")
+        else:
+            js9status.write("not a valid repo")
+            bye("--js9 {options.js9}: not a valid directory or git repository")
+    elif os.path.isdir(JS9_DIR):
+        js9_dir = JS9_DIR
+        message("Will install JS9 from default directory {js9_dir}")
+    else:
+        js9_git = JS9_REPO
+        message("{JS9_DIR} not found. Will clone JS9 from default location ({JS9_REPO})")
+
+    # Run git clone if needed
+    if js9_git:
+        if os.access(PADRE_PATH, os.W_OK) and not os.path.exists(JS9_DIR):
+            js9_dir = JS9_DIR
+        else:
+            import tempfile
+            js9_dir = js9_temp = tempfile.mkdtemp(prefix="js9")
+        message("Cloning {js9_git} into {js9_dir}")
+        if shell("git clone {js9_git} {js9_dir}") != 0:
+            js9status.write("git clone failed")
+            bye("Failed to clone JS9 repo, see log above")
+    else:
+        if js9_dir == JS9_DIR and os.access(JS9_DIR, os.W_OK) and os.path.exists(JS9_DIR + "/.git"):
+            if not options.skip_git_pull:
+                message("Will do a git pull in {js9_dir}")
+                if shell("git -C {js9_dir} pull") != 0:
+                    js9status.write("git pull failed")
+                    bye("Failed to git pull in JS9 directory, see log above")
+
+    # Configure and install
+    js9_www = PADRE_VENV + "/js9-www"
+    if subprocess.call("""cd {js9_dir} && \
+            ./configure --prefix={PADRE_VENV} --with-webdir={js9_www} --with-helper=nod --with-cfitsio={options.cfitsio_path} && \
```

Comment:
@o-smirnov is it supposed to be `--with-helper=nodejs`?

Reply:
Yes it ought to say ``nodejs``. Dunno how this built at all -- maybe @ericmandel is not so strict on checking the value....